metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeExtractor.scala

```diff
@@ -31,10 +26,9 @@ final class FoldingRangeExtractor(
   }
 
   def extractFrom(tree: Tree, enclosing: Position): Unit = {
-    if (span(tree.pos) > spanThreshold) {
+    if (span(tree.pos) > 0) {
       val newEnclosing = tree match {
-        case Foldable((pos, adjust))
-            if span(enclosing) - span(pos) > distanceToEnclosingThreshold =>
+        case Foldable((pos, adjust)) =>
```

Comment:
Note: It's probably not neccessary to limit the amount of foldable regions as users might want to fold different things anywa.

Btw. @MaciejG604 make sure that we don't add multiple at the same line.